src/zoauthServer.js

```diff
@@ -373,6 +374,10 @@ export class ZOAuthServer {
     } else if (
       ZOAuthServer.validateCredentialsValue(username, email, password, policies)
     ) {
+      if (!extras.accept) {
+        response.result = { error: "Please accept policies's terms" };
+        return response;
```

Comment:
As we said, maybe this function need refactoring. Early return strategy doesn't seems to be used